src/com/twitter/intellij/pants/ui/PantsConsoleViewPanel.java

```diff
@@ -0,0 +1,158 @@
+    myProject = project;
+    myConfiguration = ErrorTreeViewConfiguration.getInstance(project);
+    setLayout(new BorderLayout());
+
+    myAutoScrollToSourceHandler = new AutoScrollToSourceHandler() {
+      @Override
+      protected boolean isAutoScrollMode() {
+        return myConfiguration.isAutoscrollToSource();
+      }
+
+      @Override
+      protected void setAutoScrollMode(boolean state) {
+        myConfiguration.setAutoscrollToSource(state);
+      }
+    };
+
+    myMessagePanel = new JPanel(new BorderLayout());
+
+    DefaultMutableTreeNode root = new DefaultMutableTreeNode();
+    final DefaultTreeModel treeModel = new DefaultTreeModel(root);
+    myTree = createTree(treeModel);
+    myTree.getEmptyText().setText(IdeBundle.message("errortree.noMessages"));
+
+    myAutoScrollToSourceHandler.install(myTree);
+    TreeUtil.installActions(myTree);
+    UIUtil.setLineStyleAngled(myTree);
+    myTree.setRootVisible(false);
+    myTree.setShowsRootHandles(true);
+    myTree.setLargeModel(true);
+
+    JScrollPane scrollPane = NewErrorTreeRenderer.install(myTree);
+    scrollPane.setBorder(IdeBorderFactory.createBorder(SideBorder.LEFT));
+    myMessagePanel.add(PantsConsoleManager.getOrMakeNewConsole(myProject).getComponent(), BorderLayout.CENTER);
+
+    add(createToolbarPanel(), BorderLayout.WEST);
+
+    add(myMessagePanel, BorderLayout.CENTER);
+
+    EditSourceOnDoubleClickHandler.install(myTree);
+  }
+
+  @NotNull
+  protected Tree createTree(@NotNull final DefaultTreeModel treeModel) {
+    return new Tree(treeModel) {
+      @Override
+      public void setRowHeight(int i) {
+        super.setRowHeight(0);
+        // this is needed in order to make UI calculate the height for each particular row
+      }
+    };
+  }
+
+  public class PantsProcessCancellationAction extends AnAction implements DumbAware {
+    public PantsProcessCancellationAction() {
+      copyFrom(ActionManager.getInstance().getAction(IdeActions.ACTION_CLOSE_ACTIVE_TAB));
+      Presentation presentation = getTemplatePresentation();
+      presentation.setIcon(AllIcons.Actions.Suspend);
+      presentation.setText(CommonBundle.getCloseButtonText());
+      presentation.setDescription(null);
+    }
+
+    @Override
+    public void actionPerformed(AnActionEvent e) {
+      PantsMakeBeforeRun.terminatePantsProcess(e.getProject());
+      PantsConsoleManager.getOrMakeNewConsole(e.getProject()).print("Pants process terminated.", ConsoleViewContentType.ERROR_OUTPUT);
+    }
+  }
+
+  private class StopAction extends DumbAwareAction {
+    public StopAction() {
+      super(IdeBundle.message("action.stop"), null, AllIcons.Actions.Suspend);
+    }
+
+    @Override
+    public void actionPerformed(AnActionEvent e) {
+      PantsMakeBeforeRun.terminatePantsProcess(e.getProject());
+      PantsConsoleManager.getOrMakeNewConsole(e.getProject()).print("\nPants process terminated.", ConsoleViewContentType.ERROR_OUTPUT);
```

Comment:
same. It also feels like PantsProcessCancellationAction and StopAction could have this extracted to a common location. Unless there's a reason why they are slightly different.